src/nvim/option.c

```diff
@@ -7510,7 +7511,16 @@ bool tabstop_set(char_u *var, long **array)
 
   t = 1;
   for (cp = var; *cp != NUL;) {
-    (*array)[t++] = atoi((char *)cp);
+    int n = atoi((char *)cp);
+
+    // Catch negative values, overflow and ridiculous big values.
+    if (n < 0 || n > 9999) {
+      semsg(_(e_invarg2), cp);
+      xfree(*array);
+      *array = NULL;
```

Comment:
bit of a nit actually but you could use `XFREE_CLEAR` here instead

```suggestion
      XFREE_CLEAR(*array);
```

Reply:
Yeah, I did not understand why upstream was not using `VIM_CLEAR`...

Reply:
feel free to add it and i'll merge :+1: